server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestDeleteSnapshotAction.java

```diff
@@ -29,6 +32,9 @@
 @ServerlessScope(Scope.INTERNAL)
 public class RestDeleteSnapshotAction extends BaseRestHandler {
 
+    private static final Set<String> SUPPORTED_QUERY_PARAMETERS = Set.of(RestUtils.REST_MASTER_TIMEOUT_PARAM, "wait_for_completion");
+    private static final Set<String> ALL_SUPPORTED_PARAMETERS = Sets.union(SUPPORTED_QUERY_PARAMETERS, Set.of("repository", "snapshot"));
```

Comment:
Figured it made sense to generate these once and hold on to them? Could wrap the `Sets.union` in a `Collections.unmodifiableSet` if we want to be defensive. I wonder if we want to make `"wait_for_completion"`/`"repository"`/`"snapshot"` constants here now they're used in a couple of places.

If you implement `supportedQueryParameters` you need to implement `allSupportedParameters`